pkg/lakom/config/config.go

```diff
@@ -87,9 +87,52 @@ func (c *Config) Complete() (*CompletedConfig, error) {
 		res.Keys = append(res.Keys, config)
 	}
 
+	res.Keys = uniqueKeys(res.Keys)
+
 	return &res, nil
 }
 
+type PK interface {
+	Equal(x crypto.PublicKey) bool
+}
+
+// uniqueKeys returns a new VerifierKey array that contains only the unique
+// keys from 'keys'.
+//
+// Equality between the keys is achieved via the 'Equal' function that we know
+// is implemented for every public key type. Ref: https://pkg.go.dev/crypto#PublicKey
+// Thus, casting all the keys to the PK interface enables us to call the function.
+func uniqueKeys(keys []VerifierKey) []VerifierKey {
+	var res []VerifierKey
+	var idxUniq []int
+
+	var keysCast []PK
+	for _, k := range keys {
+		keysCast = append(keysCast, k.Key.(PK))
+	}
+
+	for idx, k1 := range keysCast {
+		uniq := true
+		for _, uniqIdx := range idxUniq {
+			k2 := keysCast[uniqIdx]
+			if k1.Equal(k2) {
+				uniq = false
+				break
+			}
+		}
+
+		if uniq {
+			idxUniq = append(idxUniq, idx)
+		}
+	}
+
+	for _, idx := range idxUniq {
+		res = append(res, keys[idx])
+	}
+
+	return res
+}
+
```

Comment:
My version of this function 

```suggestion
// uniqueKeys returns a new VerifierKey slice that contains only the unique
// keys from the passed keys.
func uniqueKeys(keys []VerifierKey) []VerifierKey {
	// Equality between the keys is achieved via the 'Equal' function that we know
	// is implemented for every public key type. Ref: https://pkg.go.dev/crypto#PublicKey
	// Thus, casting all the keys to the pubKey interface enables us to call the function.
	type pubKey interface {
		Equal(x crypto.PublicKey) bool
	}

	var pubKeys []pubKey
	for _, k := range keys {
		pubKeys = append(pubKeys, k.Key.(pubKey))
	}

	var (
		res     []VerifierKey
		idxUniq []int
	)
	for idx, k1 := range pubKeys {
		uniq := true
		for _, uniqIdx := range idxUniq {
			k2 := pubKeys[uniqIdx]
			if k1.Equal(k2) {
				uniq = false
				break
			}
		}

		if uniq {
			idxUniq = append(idxUniq, idx)
		}
	}

	for _, idx := range idxUniq {
		res = append(res, keys[idx])
	}

	return res
}
```

Reply:
Alternative and simpler version (IMO) of `uniqueKeys` would look like this.

``` go
// uniquekeys returns the unique set of [VerifierKey] items.
func uniqueKeys(keys []VerifierKey) []VerifierKey {
	result := make([]VerifierKey, 0)

	for _, item := range keys {
		key, ok := item.Key.(interface{ Equal(x crypto.PublicKey) bool })
		if !ok {
			continue
		}

		predicate := func(other VerifierKey) bool {
			return key.Equal(other.Key)
		}
		if !slices.ContainsFunc(result, predicate) {
			result = append(result, item)
		}
	}

	return result
}
```

Reply:
Thank you both for the improvements, I've used the approach of @dnaeon since I agree is the simplest :)